engine/src/flutter/impeller/renderer/backend/gles/blit_command_gles.cc

```diff
@@ -328,10 +353,22 @@ bool BlitCopyTextureToBufferCommandGLES::Encode(
   }
 
   DeviceBufferGLES::Cast(*destination)
-      .UpdateBufferData([&gl, this](uint8_t* data, size_t length) {
+      .UpdateBufferData([&gl, this, coord_system,
+                         rows = source->GetSize().height](uint8_t* data,
+
+                                                          size_t length) {
         gl.ReadPixels(source_region.GetX(), source_region.GetY(),
                       source_region.GetWidth(), source_region.GetHeight(),
                       GL_RGBA, GL_UNSIGNED_BYTE, data + destination_offset);
+        switch (coord_system) {
+          case TextureCoordinateSystem::kUploadFromHost:
+            break;
+          case TextureCoordinateSystem::kRenderToTexture:
+            // The texture is upside down, and must be inverted when copying
+            // byte data out.
+            FlipImage(data + destination_offset, 4, source_region.GetHeight(),
```

Comment:
These arguments aren't right. The stride goes last.

Reply:
Oh, lol. The order doesn't matter since its its multiplied. Anyway, not how I meant the arguments would be interpreted 🙃

Reply:
oops... 


But thanks to associativity it is mathematically equivalent 😆

Reply:
Fixed